src/test/java/stepDefinitions/SEDTweaksSteps.java

```diff
@@ -12,7 +26,171 @@ public SEDTweaksSteps()
+		if(driver.getPageSource().contains("No Certification Criteria were tested for SED."))
+		    System.out.println("No Certification Criteria were tested for SED.");
+		else
+			driver.getPageSource().contains("Date SED Testing was Completed");
 	
+	}
+
+	@Then("^the header should be named 'Date SED Testing was Completed'$")
+	public void the_header_should_be_named_Date_SED_Testing_was_Completed() throws Throwable {
+	    assertTrue(
+				driver.getPageSource().contains("Date SED Testing was Completed"));
+	}
+
+	@When("^I look at the download button$")
+	public void i_look_at_the_download_button() throws Throwable {
+	   driver.findElement(By.cssSelector(".panel-body > div:nth-child(1) > span:nth-child(1) > span:nth-child(5) > table:nth-child(2) > tfoot:nth-child(2) > tr:nth-child(1) > td:nth-child(1) > button:nth-child(1)")).getTagName();
+	
+	}
+
+	@Then("^the button should be named 'Download SED Details'$")
+	public void the_button_should_be_named_Download_SED_Details() throws Throwable {
+		assertTrue(
+				driver.getPageSource().contains("Download SED Details"));
+	}
+
+	@When("^I look at the associated criteria$")
+	public void i_look_at_the_associated_criteria() throws Throwable {
+		driver.get("https://chpl.ahrqstg.org/#/product/8481");
+		WebDriverWait wait = new WebDriverWait(driver, 20);
+		wait.until(ExpectedConditions.presenceOfElementLocated(By.cssSelector("div.panel:nth-child(4) > div:nth-child(1) > a:nth-child(1)")));
+		
+	}
+
+	@Then("^the criteria number should also include the criteria title$")
+	public void the_criteria_number_should_also_include_the_criteria_title() throws Throwable {
+		Thread.sleep(5000);
+		WebElement expanded = driver.findElement(By.cssSelector("div.panel:nth-child(2) > div:nth-child(1) > a:nth-child(1)"));
+		expanded.click();
+					
+		WebElement expanded1 = driver.findElement(By.cssSelector("div.panel:nth-child(4) > div:nth-child(1) > a:nth-child(1)"));
+		expanded1.click();
+		
+		 WebElement Table_1 = driver.findElement(By.xpath("//*[@id='mainContent']/div[2]/div[2]/div/div/div[3]/div[2]/div/span/span[4]/table/thead/tr/th[1]"));
+		 List<WebElement> Rows = Table_1.findElements(By.tagName("tr"));
+		 System.out.println("No. of rows: "+Rows.size());
+		 
+		  
+		 //to print all the values inside the table
+		 for(WebElement Row : Rows)
+		 {
+		  List<WebElement> Cols = Row.findElements(By.tagName("li"));
+		  for (WebElement col : Cols) {
+              System.out.print(col.getText() + "\t");
+		  }
+		  System.out.println();
+		 }
+		}
+		
+	
+
+	@Given("^I am looking at listing's SED details$")
+	public void i_am_looking_at_listing_s_SED_details() throws Throwable {
+	    driver.get("https://chpl.ahrqstg.org/#/product/8482");
+	    WebDriverWait wait = new WebDriverWait(driver, 20);
+		wait.until(ExpectedConditions.presenceOfElementLocated(By.cssSelector("div.panel:nth-child(4) > div:nth-child(1) > a:nth-child(1)")));
+		
+	}
+
+	@Given("^present listing doesn't have SED$")
```

Comment:
Maybe we should use one Step for "navigate to a listing" and have that be parameterizable, and have all of the steps where we need to get to a particular type of Listing leverage that step

Reply:
Agree with the idea behind this comment. Let's consolidate the "navigate to a listing" functionality into one place that can have a parameter passed in but let's do it in the next iteration of AQA development. Next time Ashwini needs to load a listing page, figure out how to do that with parameters and circle back around to this code.